src/main/java/org/spongepowered/api/profile/GameProfileManager.java

```diff
@@ -24,201 +24,30 @@
-     * @return The result of the request
-     */
-    CompletableFuture<Map<String, Optional<GameProfile>>> getAllByName(Iterable<String> names, boolean useCache);
-
-    /**
-     * Fills a {@link GameProfile}.
-     *
-     * <p>The returned {@link CompletableFuture} throws an {@link ExecutionException}
-     * caused by a {@link ProfileNotFoundException} if the profile does not exist or
-     * an {@link IOException} if a network error occurred.</p>
-     *
-     * @param profile The profile to fill
-     * @return The result of the request
-     */
-    default CompletableFuture<GameProfile> fill(GameProfile profile) {
-        return this.fill(profile, false);
-    }
+public interface GameProfileManager extends GameProfileProvider {
 
     /**
-     * Fills a {@link GameProfile}.
+     * Gets the {@link GameProfileCache} of this provider which will be
+     * used to cache game profiles.
      *
-     * <p>The returned {@link CompletableFuture} throws an {@link ExecutionException}
-     * caused by a {@link ProfileNotFoundException} if the profile does not exist or
-     * an {@link IOException} if a network error occurred.</p>
+     * <p>In vanilla, properties do not persist between restarts and profiles
+     * will expire after a month.</p>
      *
-     * @param profile The profile to fill
-     * @param signed true if we should request that the profile data be signed
-     * @return The result of the request
-     */
-    default CompletableFuture<GameProfile> fill(GameProfile profile, boolean signed) {
-        return this.fill(profile, signed, true);
-    }
-
-    /**
-     * Fills a {@link GameProfile}.
-     *
-     * <p>The returned {@link CompletableFuture} throws an {@link ExecutionException}
-     * caused by a {@link ProfileNotFoundException} if the profile does not exist or
-     * an {@link IOException} if a network error occurred.</p>
-     *
-     * @param profile The profile to fill
-     * @param signed true if we should request that the profile data be signed
-     * @param useCache true to perform a cache lookup first
-     * @return The result of the request
-     */
-    CompletableFuture<GameProfile> fill(GameProfile profile, boolean signed, boolean useCache);
-
-    /**
-     * Gets the active {@link GameProfile} cache.
-     *
-     * @return The active cache
+     * @return The cache
      */
     GameProfileCache getCache();
 
     /**
-     * Sets the {@link GameProfile} cache.
-     *
-     * <p>To restore the original cache, pass the result of {@link #getDefaultCache()}.</p>
+     * Gets a {@link GameProfileProvider} which bypasses the
+     * {@link GameProfileCache}.
      *
-     * @param cache The new cache
+     * @return The uncached game profile provider
      */
-    void setCache(GameProfileCache cache);
-
-    /**
-     * Gets the default cache.
-     *
-     * @return The default cache.
-     */
-    GameProfileCache getDefaultCache();
-
+    GameProfileProvider uncached();
```

Comment:
getUncached?

getDynamic?

With that said, I am not sure how wise it is to expose a provider that doesn't cache. As only due to the fact that profiles can be slow to query and Mojang does rate limit an IP from querying so many per minute I think.

Reply:
The uncached one can still cache to consider rate limits, it just bypasses the default cache which retains profiles for a month.